doc/engsys_resource_management.md

```diff
@@ -0,0 +1,94 @@
+## Resource management guidelines
+
+This document contains guidelines for creating and managing Azure resources in the Azure SDK Engineering System
+subscriptions. EngSys has automation that will delete resources based on the criteria outlined below. This is to
+improve our overall security and keep us in compliance with company policies.
+
+  * [Managing Dev Resources in the Playground Subscription](#managing-dev-resources-in-the-playground-subscription)
+     * [Resource Groups](#resource-groups)
+     * [Role Assignments](#role-assignments)
+  * [Managing CI resources in the test subscription](#managing-ci-resources-in-the-test-subscription)
+     * [Resource Groups](#resource-groups-1)
+     * [Role Assignments](#role-assignments-1)
+
+
+## Managing Dev Resources in the Playground Subscription
+
+This section applies to resource groups located in the `Azure SDK Developer Playground` subscription.
+
+Currently the automation will inspect resource groups and role assignments only.
+
+### Resource Groups
+
+The automation will delete all resource groups in the playground subscription DAILY, unless they meet one of the following criteria:
+
+- The resource group starts with a valid Microsoft alias within `microsoft.com` or `ntdev.microsoft.com`, followed by an
+  optional hyphen and extra characters.
+    - Valid group name examples: `myalias`, `myalias-test-foobar` for `myalias@microsoft.com`
+- The resource group contains a tag with the name `owners` where the value is a csv formatted string that contains at
+  least one valid Microsoft alias within `microsoft.com` or `ntdev.microsoft.com`. This convention should only be used
+  when it is not possible to name the resource group with an alias (for example, inner groups auto-created by a resource
+  provider like AKS).
+    - Valid owner tag examples: `owners: myalias`, `owners: myalias,anotheralias,lastalias`
+- The resource group contains a tag with the name `DeleteAfter` and an ISO8601 formatted date value, where the date is
+  not greater than 7 days in the future.
+    - If the `DeleteAfter` value is in the past, or greater than 7 days in the future, it will be deleted.
```

Comment:
Should we clarify the unit of sensitivity here?  If someone uses a date/time value that is "today at 11am" and the cleanup script runs at 12pm, does that count as "in the past" or is it only considering the date component and ignoring time?

If the time is significant, we may want to guarantee a runtime window for the script.   Maybe something like "the script will run no earlier than 12pm PT to allow time to extend resource groups that expire today?"

Reply:
That's an interesting point. Currently the pipeline is scheduled to run every 4 hours, but if we ever change that schedule I think it's guaranteed to lead to documentation rot. Alternatively, we could have multiple iterations of the cleanup pipeline, one that runs for playground with a daily schedule and one that runs more aggressively for the automated resources test sub. @weshaggard thoughts?

Reply:
I would keep the same automation for both and running it more often seems like it should be fine. I think we should simply harden the guidelines to require a time and if no time is specified then it will have the default of 12PM which means the first time the clean-up runs on that day it will clean-up that resource.

Reply:
Agreed, except the schedule starts at midnight, not noon.